pkg/postage/batchstore/reserve_test.go

```diff
@@ -0,0 +1,930 @@
+				batchUnreserve(2, 5),
+				batchUnreserve(3, 5),
+				batchUnreserve(4, 5),
+			},
+		}, {
+			// dilute the batch at value 3, expect to evict out the
+			// batch with value 4 to radius 5
+			desc:   "dilute cheaper batch and evict batch from outer",
+			dilute: []batchDepthTuple{batchDepth(1, 9)},
+			exp: []batchUnreserveTuple{
+				batchUnreserve(0, 5),
+				batchUnreserve(1, 5),
+				batchUnreserve(2, 5),
+				batchUnreserve(3, 4),
+				batchUnreserve(4, 4),
+			},
+		}, {
+			// top up the highest value batch to be value 12, then dilute it
+			// to be depth 9 (original 8), which causes it to be at value 6
+			// expect batches with value 4 and 5 to evict outer, and the last
+			// batch to be at outer tier (radius 4)
+			// inner 2, outer 6
+			desc:   "dilute cheaper batch and evict batch from outer",
+			topup:  []batchValueTuple{batchValue(4, 12)},
+			dilute: []batchDepthTuple{batchDepth(4, 9)},
+			exp: []batchUnreserveTuple{
+				batchUnreserve(0, 5),
+				batchUnreserve(1, 5),
+				batchUnreserve(2, 5),
+				batchUnreserve(3, 5),
+				batchUnreserve(4, 4),
+			},
+		},
+	} {
+		t.Run(tc.desc, func(t *testing.T) {
+			store, unreserved := setupBatchStore(t)
+			batches := addBatch(t, store,
+				depthValue(initBatchDepth, 2),
+				depthValue(initBatchDepth, 3),
+				depthValue(initBatchDepth, 4),
+				depthValue(initBatchDepth, 5),
+				depthValue(initBatchDepth, 6),
+			)
+
+			topupBatch(t, store, batches, tc.topup...)
+			diluteBatch(t, store, batches, tc.dilute...)
+
+			for _, v := range tc.exp {
+				b := []*postage.Batch{batches[v.batchIndex]}
+				checkUnreserved(t, unreserved, b, v.expDepth)
+			}
+		})
+	}
+}
+
+func TestBatchStore_EvictExpired(t *testing.T) {
+	// temporarily reset reserve Capacity
+	defer func(i int64) {
+		batchstore.Capacity = i
+	}(batchstore.Capacity)
+	batchstore.Capacity = batchstore.Exp2(5) // 32 chunks
+	initBatchDepth := uint8(8)
+
+	store, unreserved := setupBatchStore(t)
+	batches := addBatch(t, store,
+		depthValue(initBatchDepth, 2),
+		depthValue(initBatchDepth, 3),
+		depthValue(initBatchDepth, 4),
+		depthValue(initBatchDepth, 5),
+	)
+
+	cs := store.GetChainState()
+	cs.Block = 4
+	cs.Total = big.NewInt(4)
+	err := store.PutChainState(cs)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// expect the 5 to be preserved and the rest to be unreserved
+	checkUnreserved(t, unreserved, batches[:3], swarm.MaxPO)
+	checkUnreserved(t, unreserved, batches[3:], 4)
+
+	// check that the batches is actually deleted from
+	// statestore, by trying to do a Get on the deleted
+	// batches, and assert that they are not found
+	for _, v := range batches[:3] {
+		_, err := store.Get(v.ID)
+		if !errors.Is(err, storage.ErrNotFound) {
+			t.Fatalf("expected err not found but got %v", err)
+		}
+	}
+}
+
+type depthValueTuple struct {
+	depth uint8
+	value int
+}
+
+func depthValue(d uint8, v int) depthValueTuple {
+	return depthValueTuple{depth: d, value: v}
+}
+
+type batchValueTuple struct {
```

Comment:
you can just remove 'Tuple' fromall names

Reply:
but that would force me to rename the function calls to create the instances to be renamed in a different way, since the same name of the type cannot be redeclared as a function (that's the reason why i use this naming convention)